cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/LocalApplicationTemplateManager.java

```diff
@@ -112,7 +112,7 @@ public LocalApplicationTemplateManager(CConfiguration configuration, PipelineFac
                                                     queueAdmin, discoveryServiceClient));
     pipeline.addLast(new ProgramGenerationStage(configuration, namespacedLocationFactory));
     pipeline.addLast(new ApplicationRegistrationStage(store));
-    pipeline.setFinally(new EnableConcurrentRunsStage(preferencesStore));
+    pipeline.addLast(new EnableConcurrentRunsStage(preferencesStore));
```

Comment:
sound out setFinally is like the finally block in a try catch statement, which was used to do pipeline cleanup. But some time ago I moved cleanup into the stage that causes the mess, but didn't change the setFinally